etc/make_small_lib.pl

```diff
@@ -0,0 +1,335 @@
+#!/usr/bin/env perl
+
+use strict;
+use warnings;
+
+use Getopt::Long;
+use File::Basename 'basename';
+use File::Glob 'bsd_glob';
+use File::Spec;
+
+my $sep = File::Spec->catfile('', '');
+my %fast_functions = (
+       s_mp_balance_mul            => 1,
+       s_mp_exptmod_fast           => 1,
+       s_mp_invmod_fast            => 1,
+       s_mp_karatsuba_mul          => 1,
+       s_mp_karatsuba_sqr          => 1,
+       s_mp_montgomery_reduce_fast => 1,
+       s_mp_mul_digs_fast          => 1,
+       s_mp_mul_high_digs_fast     => 1,
+       s_mp_sqr_fast               => 1,
+       s_mp_toom_mul               => 1,
+       s_mp_toom_sqr               => 1
+   );
+my @dependency_list = ();
+
+sub uniq {
+    my %seen;
+    grep !$seen{$_}++, @_;
+}
+
+sub read_file {
+  my $f = shift;
+  open my $fh, "<", $f or die "FATAL: read_rawfile() cannot open file '$f': $!";
+  binmode $fh;
+  return do { local $/; <$fh> };
+}
+
+sub write_file {
+  my ($f, $data) = @_;
+  die "FATAL: write_file() no data" unless defined $data;
+  open my $fh, ">", $f or die "FATAL: write_file() cannot open file '$f': $!";
+  binmode $fh;
+  print $fh $data or die "FATAL: write_file() cannot write to '$f': $!";
+  close $fh or die "FATAL: write_file() cannot close '$f': $!";
+  return;
+}
```

Comment:
File::Slurper...come on, it should be used because of the name already

Reply:
While we are at it: this one is harmless but others are not. You might look up all pronounceable names at urbandictionary.com first. (Example: `slong` for `signed long`).

Reply:
?

Reply:
> ?

There was a regime change here at Github, forgot?
This includes a now significant chance that somebody (e.g.: one of the shareholders) complains and another somebody (e.g.: an intern at MS) runs a badly written script over the sources here.
I don't think that it will get as bad as the rotten mess that currently happens over at YT but I also don't want to learn the differences between github and e.g.: gitlab.

But serious: no unintended(!) giggles, please, thank you.